source/elements/oneDAL/include/oneapi/dal/array.hpp

```diff
@@ -0,0 +1,290 @@
+
+    /// Creates a new array instance by passing the pointer to externally-allocated memory block
+    /// for mutable data. It is responsibility of the calling application to free memory block
+    /// as the array shall not free it when reference counter equals to zero.
+    ///
+    /// @param data         The pointer to externally-allocated memory block.
+    /// @param count        The number of elements of type $Data$ in the memory block.
+    /// @param dependencies Events indicating availability of the $data$ for reading or writing.
+    /// @pre :expr:`data != nullptr`
+    /// @pre :expr:`count > 0`
+    static array<Data> wrap(Data* data,
+                            std::int64_t count,
+                            const sycl::vector_class<sycl::event>& dependencies = {});
+
+    /// Creates a new array instance by passing the pointer to externally-allocated memory block
+    /// for immutable data. It is responsibility of the calling application to free memory block
+    /// as the array shall not free it when reference counter equals to zero.
+    ///
+    /// @param data         The pointer to externally-allocated memory block.
+    /// @param count        The number of elements of type $Data$ in the memory block.
+    /// @param dependencies Events indicating availability of the $data$ for reading or writing.
+    /// @pre :expr:`data != nullptr`
+    /// @pre :expr:`count > 0`
+    static array<Data> wrap(const Data* data,
+                            std::int64_t count,
+                            const sycl::vector_class<sycl::event>& dependencies = {});
+
+public:
+    /// Creates a new instance of the class without memory allocation:
+    /// :expr:`mutable_data` and :expr:`data` pointers shall be set to ``nullptr``,
+    /// :expr:`count` shall be to zero; the pointer to the ownership structure shall be set to ``nullptr``.
+    array();
+
+    /// Creates new array instance that shares an ownership with $other$ on its memory block.
+    array(const array<Data>& other);
+
+    /// Moves :expr:`data`, :expr:`mutable_data` pointers, :expr:`count`, and pointer to the ownership structure
+    /// in $other$ to the new array instance
+    array(array<Data>&& other);
+
+    /// Creates a new array instance which owns a memory block of externally-allocated mutable data.
+    /// The ownership structure shall be created for a block, the input $deleter$
+    /// shall be assigned to it.
+    ///
+    /// @tparam Deleter     The type of deleter used to free the $data$.
+    ///                     The deleter shall provide ``void operator()(Data*)`` member function.
+    ///
+    /// @param queue        The SYCL* queue object.
+    /// @param data         The pointer to externally-allocated memory block.
+    /// @param count        The number of elements of type $Data$ in the memory block.
+    /// @param deleter      The object used to free $data$ when reference
+    ///                     count on the memory block becomes zero.
```

Comment:
I'd use "reaches zero" or "is zero" instead of "becomes zero"

applies everywhere else as well